sktime/benchmarking/_base_kotsu.py

```diff
@@ -0,0 +1,134 @@
+"""Interface to overide kotsu code for sktime use."""
+
+import re
+import warnings
+from typing import Callable, Optional, Union
+
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("kotsu", severity="error")
```

Comment:
for soft dependency isolation, there should be no error or warning throwing statement from soft deps outside functions or classes.

A workaround for the situation below is to have an if-else statement, and if `kotsu` is not there, define empty classes of the same name. Not the most elegant, but works without having to resort to wrapping or dynamic inheritance.

Reply:
Thank you. I don't think there's a need to create empty classes because the file is only executed within the `BaseBenchmark `class, which already checks for kotsu dependencies. If kotsu does not exist in the environment, it will raise an error message so this file is not executed. 

I added the if statement in the recent commits. kindly review them.

Reply:
yes, that solves it, althouhg that introduced another level of indentation. The "dummy class" solution would have avoided that, but it's not a big difference.